### PR TITLE
removed all console.logs, now using winston.

### LINKED ISSUE #12: error handling
- [x] store errors in const file
 - [x] log errors
 - [x] return errors to client
 - [x] use proper status codes

### PR DESCRIPTION
**Pull request by zvikarp**
this is the last part of error handling, logging the errors. closes #12 .

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,24 +9,26 @@ const http = require("http");
 const { PORT, MONGODB_URI } = rootRequire("config");
```
Comment (itamar-saraf) [inline index.js]: Did you try this?
https://stackoverflow.com/questions/47540962/vscode-compatible-with-rootrequire

File: utils/winston.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const winston = require("winston");
```
Comment (itamar-saraf) [inline utils/winston.js]: where Winston outputs its logs?

Comment (zvikarp) [reply utils/winston.js]: currently only in the console. we can save it to a file, to a db or whatever. i didn't know where and if we need to save it at this stage so just logging to the console.

Comment (zvikarp) [reply utils/winston.js]: you can see on line 12 that there aren't any transports set (=places to save the log) but on line 16 we add the logging to the console if not in prod.

File: index.js (hunk 1 above)
Comment (zvikarp) [reply index.js]: cant get it to work

**Comment by zvikarp** [comment]
can we merge this?

File: index.js (hunk 1 above)
Comment (itamar-saraf) [reply index.js]: So, f12 doesn't work anymore?

Comment (zvikarp) [reply index.js]: correct, but it is not connected to this pr, it has its own issue #10

**Pull request merged**